packages/util/src/graph2.ts

```diff
@@ -437,4 +437,103 @@ export class Graph2<V = any, E = any, S = any> {
         }
         return retVal;
     }
+
+    dijkstra(source: string, target: string): { ids: string[], len: number } {
+        const edges = this.edges();
+        const Q = new Set<string>();
+        const prev: { [key: string]: string } = {};
+        const dist: { [key: string]: number } = {};
+        const adj: { [key: string]: { [key: string]: number } } = {};
+
+        function vertex_with_min_dist(Q: Set<string>, dist: { [key: string]: number }) {
+            let min_distance = Infinity;
+            let u: string | null = null;
+
+            Q.forEach(v => {
+                if (dist[v] < min_distance) {
+                    min_distance = dist[v];
+                    u = v;
+                }
+            });
+            return u;
+        }
+
+        for (let i = 0; i < edges.length; i++) {
+            const v1 = this._sourceFunc(edges[i]);
+            const v2 = this._targetFunc(edges[i]);
+            const len = 1;
+
+            Q.add(v1);
+            Q.add(v2);
+
+            dist[v1] = Infinity;
+            dist[v2] = Infinity;
+
+            if (adj[v1] === undefined) adj[v1] = {};
+            if (adj[v2] === undefined) adj[v2] = {};
+
+            adj[v1][v2] = len;
+            adj[v2][v1] = len;
```

Comment:
It might be a good idea to optionally allow `len` to be data-driven. Ex: Two people connected via a crime might be a shorter length than two people connected via a business.

Reply:
Agree - but will need to add "weight" support to the entire container first...